doc/lsst.pipe.base/creating-a-task.rst

```diff
@@ -0,0 +1,350 @@
+.. See \ref baseDebug for more information about debug variables, including how to specify them while running a command-line task.
+
+.. _creating-a-task-docs:
+
+Documentation
+=============
+
+To be written.
+
+.. _creating-a-task-subtasks:
+
+Subtasks
+========
+
+Each subtask is specified in the configuration as a field of type `lsst.pex.config.ConfigurableField` or (less commonly) `lsst.pex.config.RegistryField`.
+There are advantages to each:
+
+- `lsst.pex.config.ConfigurableField` advantages:
+
+  - It is easier for the user to override settings of the subtask; simply use dotted name notation:
+
+    .. code-block:: python
+
+       config.configurableSubtask.subtaskParam1 = ...
+
+    In contrast, to override configuration for a subtask specified as an `lsst.pex.config.RegistryField` you must either specify the name of the subtask to configure:
+
+    .. code-block:: python
+
+       config.registrySubtask[nameOfSelectedSubtask].subtaskParam1 = ...
+
+    Or use the `active` attribute to modify the configuration of the currently selected (active) subtask:
+
+    .. code-block:: python
+
+       config.registrySubtask.active.subtaskParam1 = ...
+
+- `lsst.pex.config.RegistryField` advantages:
+
+  - You can specify overrides for any registered subtask and they are remembered if you retarget subtasks.
+    In comparison if the subtask is specified as an `lsst.pex.config.ConfigurableField` then you can only override parameters for the currently retargeted subtask, and all overrides are lost each time you retarget.
+    Thus using an `lsst.pex.config.RegistryField` offers the opportunity to specify suitable overrides for more than one variant subtask, making it safer for the user to use those variants.
+    Of course this can get out of hand if there are many variants, so users should not assume that all variants have suitable overrides.
+
+  - Retargeting a subtask can be done using :option:`--config` on the command line, as long as the module containing the desired subtask has been imported:
+
+    .. code-block:: python
+
+       config.registrySubtask.name = "foo"
+
+    By comparison, a subtask specified as an `lsst.pex.config.ConfigurableField` can only be retargeted from a config override file (e.g. using :option:`--configfile`, never :option:`--config`):
+
+    .. code-block:: python
+
+       from ... import FooTask
+       config.configurableSubtask.retarget(FooTask)
+
+  - Variants subtasks are kept together in one registry, making it easier to find them.
+
+Our recommendation: if you anticipate that users will often wish to override the subtask with a variant, then use an `lsst.pex.config.RegistryField`.
+Otherwise use an `lsst.pex.config.ConfigurableField` to keep config overrides simpler and easier to read.
+
+For example PSF determiners and star selectors are perhaps best specified using `lsst.pex.config.RegistryField` because there are several variants users may wish to select from.
```

Comment:
I can think of ways to add syntactic sugar to `ConfigurableField` to allow command-line selection (basically another level of indirection).